### PR TITLE
Fix atmospheric scattering precision

### PR DESCRIPTION
**Pull request by adrien-de-tocqueville**
### Purpose of this PR

fix for https://fogbugz.unity3d.com/f/cases/1283957
There was some precision issues when the fog opacity was near 1
The bug doesn't seem related to the density volume, unlike what's stated in the bug, but when the camera is very close to the ground

---
### Testing status

tested the repro steps of the bug report (it is related to the camera position on the y axis and not the density volume though)
and tried different extreme value to check there is no more banding

### CODE REVIEW

**Review by iM0ve — APPROVED** [review]
I could not reproduce the issue described in FB on 10.1
But I consulted with Antoine about the risk of code change and checked HDRP template fog, that the PR does not have any impact. I think its safe to merge

Before:
image

After:
image

**Comment by adrien-de-tocqueville** [comment]
Note: the failed tests are due to PR #493, but it looks like the difference introduced then wasn't big enough to cause a failure. Now the small precision changes make it go over the threshold

**Review by sebastienlagarde — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/OpaqueAtmosphericScattering.shader [hunk 1]
```diff
@@ -118,7 +118,7 @@ Shader "Hidden/HDRP/OpaqueAtmosphericScattering"
         Pass
         {
             Cull Off    ZWrite Off
-            Blend One OneMinusSrcAlpha // Premultiplied alpha
+            Blend One SrcAlpha // Premultiplied alpha
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/OpaqueAtmosphericScattering.shader:121]: Forgot to change comment

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/OpaqueAtmosphericScattering.shader [hunk 2]
```diff
@@ -131,7 +131,7 @@ Shader "Hidden/HDRP/OpaqueAtmosphericScattering"
...
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/OpaqueAtmosphericScattering.shader:134]: And here

**Pull request merged**